components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3107,6 +3107,22 @@ where
         self.ctx.store_stat.engine_total_keys_written += metrics.written_keys;
     }
 
+    fn check_joint_state(&self, msg: &RaftCmdRequest) -> Result<()> {
+        if !self.fsm.peer.in_joint_state() {
+            return Ok(());
+        }
+        let cmd = msg.get_admin_request();
+        if cmd.has_prepare_merge() || cmd.has_commit_merge() || cmd.has_split() || cmd.has_splits()
```

Comment:
Shouldn't the pipeline checker filter them automatically?

Reply:
Do you mean epoch checker? The epoch checker can only filter the concurrent admin cmd.

Reply:
Actually both the merge and split cmd are ok. I agree that rejecting the merge cmd for simpleness. But maybe it's not necessary to reject split cmd?

Reply:
Yes, I removed the check for split cmd, PTAL again